### ISSUE TITLE
NSMutableURLRequest setValue doesn't work in some case

### ISSUE THREAD

**Issue by Seke1412** [daltoniam/SwiftHTTP#246]
I'm dealing with Nodejs + Reddis as Backend, when posting a JSON i need to addValue for header rather then use the default setValue given. Moreover, addValue method must be called before httpBody = try... to make it work. Any idea ?

```swift
public func appendParametersAsJSON(_ parameters: HTTPParameterProtocol) throws {
        addValue('application/json', forHTTPHeaderField: 'Content-Type');
        addValue('application/json', forHTTPHeaderField: 'Accept');
        if isURIParam() {
            appendParametersAsQueryString(parameters)
        } else {
            do {
                httpBody = try JSONSerialization.data(withJSONObject: parameters as AnyObject, options: JSONSerialization.WritingOptions())
            } catch let error {
                throw error
            }
            //let charset = CFStringConvertEncodingToIANACharSetName(CFStringConvertNSStringEncodingToEncoding(String.Encoding.utf8.rawValue))
            //setValue("application/json; charset=\(charset)", forHTTPHeaderField: contentTypeKey)
        }
    }
```

**Comment by acmacalister** [comment]
You need to have two `Content-Type` & `Accept` header fields on your request? Just adding headers from the outside method, as outlined in the docs doesn't work? What are you trying to accomplish?

**Comment by Seke1412** [comment]
i try to send a JSON to my Nodejs server. After several failure attempt, i found that 
`let charset = CFStringConvertEncodingToIANACharSetName(CFStringConvertNSStringEncodingToEncoding(String.Encoding.utf8.rawValue))
            setValue("application/json; charset=\(charset)", forHTTPHeaderField: contentTypeKey)`

in appendParametersAsJSON doesn't work. But replace those two lines of code with 

`addValue('application/json', forHTTPHeaderField: 'Content-Type');
        addValue('application/json', forHTTPHeaderField: 'Accept');`

will make it workout. Is it considered as a bug or something? i don't know if i'm on a right track or just my luck.

**Comment by acmacalister** [comment]
yeah, I think your issue probably the same as #248. We just pushed a fix with `2.0.1`. Try upgrading to that and I think you should be good to go.

**Comment by Seke1412** [comment]
i will try it and feedback you guys. God bless you all guys
